### PR TITLE
Create cluster refactor

### PR DESCRIPTION
**Pull request by dgoodwin**
Large refactor to tease out cluster secrets from our deployment ansible.

Adds developer deployment playbook wrapping the regular, but layering in some development specific tasks. Will now create several cluster versions by default as part of the deployment, which you can now use when creating a cluster. (previously we tried to create a version each time we create a cluster)

Migrated create cluster process to a playbook, and create all various secrets for each cluster we create. Certs are now created by ansible modules instead of our apiserver script. 

Better support for creating multiple clusters in any namespace(s).

### CODE REVIEW

**Comment by dgoodwin** [comment]
Ready for review.

File: README.md [hunk 1]
```diff
@@ -37,22 +37,19 @@
...
-To enable it, you must either set the USE_REAL_AWS variable or specify a
-real cluster-operator-ansible image to use in the ANSIBLE_IMAGE variable.
-
-  * `USE_REAL_AWS=1 contrib/examples/create-cluster.sh`
-  * `ANSIBLE_IMAGE=cluster-operator-ansible:canary contrib/examples/create-cluster.sh`
+  * `ansible-playbook contrib/ansible/create-cluster-playbook.yaml`
+    * This will create a cluster named after your username in your current context's namespace, using a fake ClusterVersion. (no actual resources will be provisioned, the Ansible image used will just verify the playbook called exists, and return indicating success)
+    * Specify `-e cluster_version` to use a real cluster version and provision an actual cluster in AWS. (see `oc get clusterversions` for list of the defaults we create)
+    * Specify `-e cluster_name`, '-e cluster_namespace`, or other variables you can override as defined at the top of the playbook.
+    * This command can be re-run to update the definition of the cluster and test how the cluster operator will respond to changed. (WARNING: do not try to change the name/namespace, as this will create a new cluster)
```
Comment (csrwng) [inline README.md]: s/respond to changed/respond to the change/

File: README.md [hunk 2]
```diff
@@ -37,22 +37,19 @@
...
-
-When using the `create-cluster.sh` script, provisioning on AWS is disabled by default.
-To enable it, you must either set the USE_REAL_AWS variable or specify a
-real cluster-operator-ansible image to use in the ANSIBLE_IMAGE variable.
-
-  * `USE_REAL_AWS=1 contrib/examples/create-cluster.sh`
-  * `ANSIBLE_IMAGE=cluster-operator-ansible:canary contrib/examples/create-cluster.sh`
+  * `ansible-playbook contrib/ansible/create-cluster-playbook.yaml`
+    * This will create a cluster named after your username in your current context's namespace, using a fake ClusterVersion. (no actual resources will be provisioned, the Ansible image used will just verify the playbook called exists, and return indicating success)
+    * Specify `-e cluster_version` to use a real cluster version and provision an actual cluster in AWS. (see `oc get clusterversions` for list of the defaults we create)
```
Comment (csrwng) [inline README.md]: oc get clusterversions -n openshift-cluster-operator

File: README.md [hunk 3]
```diff
@@ -37,22 +37,19 @@
 
   * Compile the Go code and create the Cluster Operator images (both Go and Ansible):
     * `make images`
-  * Idempotently deploy cluster operator to the OpenShift cluster.
-    * `ansible-playbook contrib/ansible/deploy-playbook.yaml`
-  * If your image changed, but the kubernetes config did not (which is usually the case), you should pods appropriately:
+  * Deploy cluster operator to the OpenShift cluster you're currently logged into:
```
Comment (csrwng) [inline README.md]: You may want to mention that you need to be a cluster admin to do this, either by using 'oc login -u system:admin' or 'oc login -u admin' (where you previously granted admin the cluster-admin role)

File: contrib/ansible/create-cluster-playbook.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+  # Variables you can optionally override from the CLI:
+  vars:
+    # NOTE: you may set 'cluster_namespace' to control where the cluster object is created. We do not
+    # set a default here as we need to lookup the current one if unset.
+
+    # ID for your cluster, defaults to current username:
+    cluster_name: "{{ lookup ('env', 'USER') }}"
+
+    # ClusterVersion to install, default to fake so we don't actually create a cluster unless user
```
Comment (csrwng) [inline contrib/ansible/create-cluster-playbook.yaml]: s/ClusterVersion to install/ClusterVersion to use/
You're not installing a ClusterVersion

File: contrib/ansible/create-cluster-playbook.yaml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    # explicitly opts-in:
+    cluster_version: origin-v3-10-fake
+
+    # Namespace where we expect cluster version to exist, does not have to match the cluster namespace
+    # and generally this can be left as is:
+    cluster_version_namespace: openshift-cluster-operator
+
+    # Section in ~/.aws/credentials with the AWS creds CO should use to create cloud infra and VMs:
```
Comment (csrwng) [inline contrib/ansible/create-cluster-playbook.yaml]: It might be useful to also allow specifying a different credentials file

**Comment by dgoodwin** [comment]
Updated following feedback from @csrwng.

**Pull request merged**